hyperlink/test/test_scheme_registration.py

```diff
@@ -0,0 +1,51 @@
+# -*- coding: utf-8 -*-
+from __future__ import unicode_literals
+
+
+from .common import HyperlinkTestCase
+from .._url import register_scheme, URL
+
+
+class TestSchemeRegistration(HyperlinkTestCase):
+
+    def test_register_scheme_basic(self):
+        register_scheme('deltron', uses_netloc=True, default_port=3030)
+
+        u1 = URL.from_text('deltron://example.com')
+        assert u1.scheme == 'deltron'
+        assert u1.port == 3030
+        assert u1.uses_netloc is True
+
+        # test netloc works even when the original gives no indication
+        u2 = URL.from_text('deltron:')
+        u2 = u2.replace(host='example.com')
+        assert u2.to_text() == 'deltron://example.com'
+
+        # test default port means no emission
+        u3 = URL.from_text('deltron://example.com:3030')
+        assert u3.to_text() == 'deltron://example.com'
+
+        register_scheme('nonetron', default_port=3031)
+        u4 = URL(scheme='nonetron')
+        u4 = u4.replace(host='example.com')
+        assert u4.to_text() == 'nonetron://example.com'
+
+        register_scheme('noloctron', uses_netloc=False)
+        u4 = URL(scheme='noloctron')
+        u4 = u4.replace(path=("example", "path"))
+        assert u4.to_text() == 'noloctron:example/path'
+
+    def test_register_no_netloc_scheme(self):
+        register_scheme('netlocless', uses_netloc=False)
```

Comment:
Please move the assertions and code in `test_register_scheme_basic` that has `uses_netloc=False` (lines 33-36) here.